src/views/components/input/file/InputFile.native.js

```diff
@@ -1,49 +1,208 @@
 import React, { Component } from 'react';
-import Expo, { ImagePicker } from 'expo';
-import { View, Image, Button } from 'react-native';
+import axios from 'axios';
+import Expo, { ImagePicker, DocumentPicker } from 'expo';
+import { bool, func } from 'prop-types';
+import dlv from 'dlv';
+import fastXmlParser from 'fast-xml-parser';
+import mime from 'react-native-mime-types';
+import { Box } from '../../../components';
+import InputFileItem from './file-item';
+import InputFileTouchable from './file-touchable';
 
 class InputFile extends Component {
   static defaultProps = {
+    imageOnly: false,
   }
 
   static propTypes = {
+    imageOnly: bool,
+    onChange: func,
   }
   
   state = {
-    image: null,
+    files: [],
   };
 
+  handleComplete = result => {
+    this.setState( prevState => ({
+      files: [
+        ...prevState.files,
+        result,
+      ],
+    }), () => {
+      this.handleSaveToServer;
+    });
+  }
+
+  handleSaveToServer = () => {
+    const { files } = this.state;
+    
+    if ( this.props.onChange ) {
+      this.props.onChange({ target: { value: files } });
+    }
+  }
+
+  uploadFile = result => {
+    const localUri = result.uri;
+    const filename = localUri.split( '/' ).pop();
+    const match = /\.(\w+)$/.exec( filename );
+
+    const type = match ? mime.contentType( match[1] ).split( ';' )[0] : 'file';  
+    const formData = new FormData();
+    const url = 'https://uppych40.channel40.com.au/s3/';
+    
+    axios({
+      method: 'get',
+      url: `${url}params?filename=${filename}&type=${type.split( '/' )[0]}%2F${type.split( '/' )[1]}`,
+    })
+    .then( response => {
+      if ( response.status === 200 ) {
+        if ( dlv( response, 'data.fields' )) {
+          const data = response.data;
+          const fields = response.data.fields;
+          
+          Object.keys( fields ).forEach( field_key => {
+            formData.append( field_key, fields[field_key] );
+          });
+
+          formData.append( 'file', { uri: localUri, name: filename, type });
+          
+          fetch( response.data.url, {
+            method: 'POST',
+            body: formData,
+            header: {
+              'content-type': 'multipart/form-data',
+            },
+          })
+          .then( async response => {
+            const text = await response.text();
+
+            const jsonObj = fastXmlParser.parse( text );
+            
+            const formattedFile = {
+              data: {
+                name: formData._parts.filter( field => ( field[0] === 'file' ))[0][1].name,
```

Comment:
Instead of doing `formData._parts.filter( field => ( field[0] === 'file' ))[0][1].name` 6 times in this function can we please do it once and assign it to a variable so it doesn't need to compute the same thing so many times?